denote.el

```diff
@@ -2508,23 +2510,19 @@ If more than one file type correspond to this file extension, use the
 first file type for which the :title-key-regexp in `denote-file-types'
 matches in the file.
 
-If no file type in `denote-file-types' has the file extension,
-the file type is assumed to be the first one in `denote-file-types'."
+Return nil if the file type is not recognized."
```

Comment:
I think the reason we did this was because we ultimately needed some way to, e.g., format links. Org links can work anywhere in Emacs with `org-open-at-point-global`.

Reply:
You mean that a non-org file can create links of org type? Right now (and also before this PR), `denote-link` would not allow that anyway. I think if we want to allow that eventually, we should add this support in the linking commands directly (and keep `denote-filetype-heuristics` simply returning the type of the file given to it in its parameter).